src/hotspot/share/services/virtualMemoryTracker.hpp

```diff
@@ -371,25 +374,107 @@ class VirtualMemoryTracker : AllStatic {
   friend class VirtualMemoryTrackerTest;
   friend class CommittedVirtualMemoryTest;
 
- public:
-  static bool initialize(NMT_TrackingLevel level);
+  enum class Tag {
+    AddCommitted, AddReserved, RemoveUncommitted, RemoveReleased, SplitReserved
+  };
+  struct MemoryEvent {
+    using StackFlag = struct {
+        NativeCallStack stack;
+        MEMFLAGS flag;
+    };
+    using Split = struct {
+      size_t split;
+    };
+    Tag tag;
+    void* addr; size_t size;
```

Comment:
Switch `addr` from `void*`  to `address` to make it clear that we don't intend to dereference this pointer.